modules/reindex/src/main/java/org/elasticsearch/index/reindex/ReindexTaskStateUpdater.java

```diff
@@ -43,31 +45,36 @@
 
     private final ReindexIndexClient reindexIndexClient;
     private final ThreadPool threadPool;
-    private final String taskId;
+    private final String persistentTaskId;
     private final long allocationId;
+    private final TaskId ephemeralTaskId;
     private final ActionListener<ReindexTaskStateDoc> finishedListener;
     private final Runnable onCheckpointAssignmentConflict;
+    private Consumer<Float> requestsPerSecondObserver;
```

Comment:
Do we need this? I feel like adding the "poll" interaction model when we are doing all the working to implement the "push" component makes the interaction with the state updater and task more complicated that required.

Reply:
Thanks, I agree, removed in b6096a9398a458273c392fcb832e140eaaa73fb2